### PR TITLE
SReLU activation

### PR DESCRIPTION
**Pull request by barvinograd**
Added the recently proposed SReLU activation. 
http://arxiv.org/abs/1512.07030
see issue #1662

### CODE REVIEW

**Comment by fchollet** [comment]
The added tests are failing.

**Comment by barvinograd** [comment]
They were failing due to inconsistency between the TensorFlow and the Theano backends. TF's ReLU did not support a non const slope for the negative part - reported as #1681. It is now fixed and all the tests are passing.

File: keras/backend/tensorflow_backend.py [hunk 1]
```diff
@@ -507,7 +507,12 @@ def relu(x, alpha=0., max_value=None):
     if max_value is not None:
         x = tf.clip_by_value(x, tf.cast(0., dtype=_FLOATX),
                              tf.cast(max_value, dtype=_FLOATX))
-    x -= tf.constant(alpha, dtype=_FLOATX) * negative_part
+    try:
+        alpha = tf.constant(alpha, dtype=_FLOATX)
+    except TypeError:
+        # in case alpha is not a const
+        pass
+    x -= alpha * negative_part
```
Comment (fchollet) [inline keras/backend/tensorflow_backend.py]: `try / except` is not the proper thing to do here. You could do a type check instead.

**Comment by barvinograd** [comment]
@fchollet thanks for the feedback. Now testing the type of `alpha`. Tests seems to be failing randomly on MNIST download. I pushed 2 commits with no code difference, one failed on [python2.7+theano](https://travis-ci.org/fchollet/keras/jobs/108515734) and the other [python3.4+tensorflow](https://travis-ci.org/fchollet/keras/jobs/108520023). Its probably temporary though.

```
==================================== ERRORS ====================================
______________ ERROR collecting tests/keras/test_regularizers.py _______________
tests/keras/test_regularizers.py:22: in <module>
    (X_train, y_train), (X_test, y_test) = mnist.load_data()
keras/datasets/mnist.py:17: in load_data
    data = cPickle.load(f)
../../../miniconda/envs/test-environment/lib/python2.7/gzip.py:464: in readline
    c = self.read(readsize)
../../../miniconda/envs/test-environment/lib/python2.7/gzip.py:268: in read
    self._read(readsize)
../../../miniconda/envs/test-environment/lib/python2.7/gzip.py:315: in _read
    self._read_eof()
../../../miniconda/envs/test-environment/lib/python2.7/gzip.py:354: in _read_eof
    hex(self.crc)))
E   IOError: CRC check failed 0x5159ec5c != 0x7a59d592L
______________ ERROR collecting tests/keras/test_regularizers.py _______________
tests/keras/test_regularizers.py:22: in <module>
    (X_train, y_train), (X_test, y_test) = mnist.load_data()
keras/datasets/mnist.py:9: in load_data
    path = get_file(path, origin="https://s3.amazonaws.com/img-datasets/mnist.pkl.gz")
keras/datasets/data_utils.py:42: in get_file
    ParanoidURLopener().retrieve(origin, fpath, dl_progress)
../../../miniconda/envs/test-environment/lib/python2.7/urllib.py:289: in retrieve
    "of %i bytes" % (read, size), result)
E   ContentTooShortError: retrieval incomplete: got only 1383448 out of 15296311 bytes
------------------------------- Captured stdout --------------------------------
Downloading data from https://s3.amazonaws.com/img-datasets/mnist.pkl.gz
 1384448/15296311 [=>............................] - ETA: 822s
```

**Comment by fchollet** [comment]
These errors appear to be connection issues on the Travis server. It doesn't matter...

**Comment by fchollet** [comment]
This looks good to me, please squash your commits and I will merge.

**Comment by barvinograd** [comment]
@fchollet done

**Pull request merged**